engines/stark/services/userinterface.cpp

```diff
@@ -264,9 +264,11 @@ const Graphics::Surface *UserInterface::getGameWindowThumbnail() const {
 }
 
 void UserInterface::onScreenChanged() {
-	_mainMenuScreen->onScreenChanged();
 	_gameScreen->onScreenChanged();
-	_diaryIndexScreen->onScreenChanged();
+
+	if (!isInGameScreen()) {
+		_currentScreen->onScreenChanged();
```

Comment:
There will be more ```Screen``` added to ```UserInterface``` later. Since ```StaticLocationScreen``` will release all the resources when it is closed, there is no need to notify the change of screen resolution for a not currently displayed screen.